src/values/stream.ts

```diff
@@ -1,30 +1,42 @@
 import {
-  ChangeFeedSuccess,
+  FeedSuccess,
   QueryValue,
   StreamEventData,
   QueryStats,
 } from "../wire-protocol";
 import { getServiceError } from "../errors";
 
 /**
- * A token used to initiate a Fauna stream at a particular snapshot in time.
+ * A token used to initiate a Fauna event source at a particular snapshot in time.
  *
- * The example below shows how to request a stream token from Fauna and use it
+ * The example below shows how to request an event token from Fauna and use it
  * to establish an event steam.
  *
  * @example
  * ```javascript
  *  const response = await client.query(fql`
  *    Messages.byRecipient(User.byId("1234"))
  *  `);
- *  const token = response.data;
+ *  const eventSource = response.data;
  *
- *  const stream = client.stream(token)
+ *  const stream = client.stream(eventSource)
  *    .on("add", (event) => console.log("New message", event))
  *
  *  stream.start();
  * ```
  */
+export interface EventSource {
+  readonly token: string;
+}
+
+export function isEventSource(value: any): value is EventSource {
+  if (typeof value.token === "string") {
+    return true;
+  }
+
+  return false;
+}
+
 export class StreamToken {
```

Comment:
Should `StreamToken` explicitly implement `EventSource` or is it sufficiently duck-typed?

Reply:
I mean... it IS duck typed. But I mean is that okay.

Reply:
I can add it explicitly. But interfaces match so TS is happy.

Reply:
When I fix all of the doc things I missed, I'll make this more explicit. I think duck typing has to be okay (and consider it a feature of interfaces), but we don't need to rely on it here.